class.ext_update.php

```diff
@@ -116,26 +91,89 @@ public function main()
+     *
+     * @return array Array of UIDs of outdated records
+     */
+    protected function getMetadataConfig(): array
+    {
+        $uids = [];
+        // check if tx_dlf_metadata.xpath exists anyhow
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_metadata');
+
+        $result = $queryBuilder
+            ->select('*')
+            ->from('tx_dlf_metadata')
+            ->execute();
+
+        $rows = $result->fetchAll();
+
+        if ((count($rows) === 0) || !array_key_exists('xpath', $rows[0])) {
+            return $uids;
+        }
+        foreach ($rows as $row) {
+            if ($row['format'] === 0 && $row['xpath']) {
+                $uids[] = (int)$row['uid'];
+            }
+        }
+        return $uids;
+    }
+
+    /**
+     * Check all configured Solr cores
+     *
+     * @access protected
+     *
+     * @return bool
+     */
+    protected function solariumSolrUpdateRequired(): bool
+    {
+        // Get all Solr cores that were not deleted.
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_solrcores');
+        $result = $queryBuilder
+            ->select('index_name')
+            ->from('tx_dlf_solrcores')
+            ->execute();
+
+        while($resArray = $result->fetch()) {
+            // Instantiate search object.
+            $solr = Solr::getInstance($resArray['index_name']);
+            if (!$solr->ready) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     /**
      * Check for old format classes
      *
      * @access protected
      *
      * @return array containing old format classes
      */
-    protected function oldFormatClasses()
+    protected function oldFormatClasses(): array
     {
         $oldRecords = [];
         // Get all records with outdated configuration.
-        $result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
-            'tx_dlf_formats.uid AS uid,tx_dlf_formats.type AS type',
-            'tx_dlf_formats',
-            'tx_dlf_formats.class IS NOT NULL AND tx_dlf_formats.class != "" AND tx_dlf_formats.class NOT LIKE "%\\\\\\\\%"' // We are looking for a single backslash...
-                . Helper::whereClause('tx_dlf_formats')
-        );
-        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_formats');
+
+        $result = $queryBuilder
+            ->select('tx_dlf_formats.uid AS uid', 'tx_dlf_formats.type AS type')
+            ->from('tx_dlf_formats')
+            ->where(
+                $queryBuilder->expr()->isNotNull('tx_dlf_formats.class'),
+                $queryBuilder->expr()->neq('tx_dlf_formats.class', $queryBuilder->createNamedParameter('')),
+                $queryBuilder->expr()->notLike('tx_dlf_formats.class', $queryBuilder->createNamedParameter('%\\\\%'))
```

Comment:
This looks better but still does not work. The below values are correct and shouldn't be touched.
```
MariaDB [typo3]> select uid,type,class from tx_dlf_formats WHERE class IS NOT NULL AND class != "" AND class NOT LIKE '%\\\\%';
+-----+--------+--------------------------------+
| uid | type   | class                          |
+-----+--------+--------------------------------+
|   1 | MODS   | Kitodo\\Dlf\\Format\\Mods      |
|   2 | TEIHDR | Kitodo\\Dlf\\Format\\TeiHeader |
|   3 | ALTO   | Kitodo\\Dlf\\Format\\Alto      |
+-----+--------+--------------------------------+
3 rows in set (0.001 sec)
```
The check should only return old formats like "tx_dlf_alto" (or how this old ALTO class was named).
Maybe @sebastian-meyer could explain the history.

Reply:
Thats weird. If I run this query, I get the following result:

`
MariaDB [kitodo9]> select uid,type,class from tx_dlf_formats WHERE class IS NOT NULL AND class != "" AND class NOT LIKE '%\\\\%';
Empty set (0.000 sec)
`

Reply:
I changed the part `NOT LIKE '%\\\\%'` to `LIKE '%tx_dlf_%'`. Now this should work as expected.